### PR TITLE
Create and use VisualOdom singleton

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
@peterbarker, this has been tested in some way I guess?

File: libraries/AP_VisualOdom/AP_VisualOdom.cpp [hunk 1]
```diff
@@ -61,6 +61,11 @@ const AP_Param::GroupInfo AP_VisualOdom::var_info[] = {
 AP_VisualOdom::AP_VisualOdom()
 {
     AP_Param::setup_object_defaults(this, var_info);
+    if (_singleton != nullptr) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "BUG: multiple VisualOdometrys not allowed");
```
Comment (rmackay9) [inline libraries/AP_VisualOdom/AP_VisualOdom.cpp]: We seem to have various messages and methods when the setup fails.

I see the AC_Fence uses a panic in SITL at least.  I haven't seen the word "BUG:" in the other ones.. wondering if we should try to be consistent with the message.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -2772,7 +2773,7 @@ void GCS_MAVLINK::handle_data_packet(mavlink_message_t *msg)
 
 void GCS_MAVLINK::handle_vision_position_delta(mavlink_message_t *msg)
 {
-    AP_VisualOdom *visual_odom = get_visual_odom();
+    AP_VisualOdom *visual_odom = AP::visualodom();
```
Comment (rmackay9) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: I think we're using get_singleton() aren't we these days?

**Comment by peterbarker** [comment]
On Thu, 14 Feb 2019, Randy Mackay wrote:

> @peterbarker, this has been tested in some way I guess?

It's testing autotest.

File: libraries/AP_VisualOdom/AP_VisualOdom.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_VisualOdom/AP_VisualOdom.cpp]: I've changed this to look like Fence.

@magicrub is fixing all of this for us :-)

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Not generally.  The access points to singletons is via the `AP::` namespace.  For the most part that's implemented with get_singleton on the relevant class.

Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: ... also, I believe I've made an even number of grammatical errors in that comment, so I'm all good.

**Comment by rmackay9** [comment]
OK, i'll merge once it's passed travis

**Pull request merged**